proto/frostdb/storage/v1alpha1/storage.proto

```diff
@@ -0,0 +1,299 @@
+syntax = "proto3";
+
+package frostdb.storage.v1alpha1;
+
+// FrostDBService is a service that allows clients to execute queries as logicalplan
+// against an instance of FrostDB.
+service FrostDBService {
+  // Query executes the query plan contained in the request and returns back
+  // the results as arrow records in IPC format.
+  rpc Query(QueryRequest) returns (QueryResponse) {}
+}
+
+// QueryRequest is the message sent to the Query gRPC endpoint.
+message QueryRequest {
+  // PlanRoot is the root of the query plan (i.e. a scan node).
+  PlanNode plan_root = 1;
+}
+
+// QueryResponse is the message received from the Query gRPC endpoint.
+message QueryResponse {
+  // The resulting arrow record as bytes.
+  bytes record = 1;
+}
```

Comment:
Similar to the above, I added more information about individual queries. One of which is always running `explain` on the query. This string is logged. 

I'm wondering what you would think about also (optionally) returning the explainer string to the caller.
```suggestion
// QueryResponse is the message received from the Query gRPC endpoint.
message QueryResponse {
  // The resulting arrow record as bytes.
  bytes record = 1;
  // explain string that shows the underlying executed physical plan
  string explain = 2;
}
```

Reply:
Like the idea, especially in the future we can have a query planner 🔝